src/fiatconnect/index.ts

```diff
@@ -184,3 +184,21 @@ export async function addNewFiatAccount(
   // TODO: use the SDK to make the request once SDK is published
   throw new Error('Not implemented')
 }
+
+/**
+ * Get an obfuscated version of a fiat account number.
+ *
+ * For most accounts this will be a series of * followed by the last 4 digits.
+ *
+ * Ensures at least 3 digits are blanked out for user privacy since it is expected that this will be used to
+ *  compute an accountName for the fiat account, which is returned in GET /accounts and thus shouldn't just
+ *  be the user's bank account number. GET /accounts is still authenticated via SIWE, so at least 3 blanked digits
+ *  should be acceptable; the obfuscation is just a secondary layer of security around sensitive info.
+ *
+ * @param accountNumber
+ */
+export function getObfuscatedAccountNumber(accountNumber: string): string {
+  const digitsToReveal = Math.max(0, Math.min(accountNumber.length - 3, 4))
+  const blanks = '*'.repeat(accountNumber.length - digitsToReveal)
```

Comment:
All of the designs I've seen just show three dots followed by last 4 digits. Was there a reason you changed to the * and made the number variable? I was imagining that if we run into schemas that have potentially 4 or less identifying numbers that we could handle it in the future on a case by case basis.

https://www.figma.com/file/bwIY7peeykI8K4wTrXs4iN/Fiat-Connect?node-id=1498%3A40640
https://www.figma.com/file/bwIY7peeykI8K4wTrXs4iN/Fiat-Connect?node-id=1895%3A38461

Reply:
(The * thing was carried over from IHL, which I tried to largely copy from for this since we've done it before there)

Reply:
this is also explained by the docstring

Reply:
also I don't agree with the strategy of punting on it-- we won't necessarily notice that we're adding geos that sometimes have short account numbers. In fact I once had an account number in the US that was just 5 digits

Reply:
Thanks, the reasoning for including truncation logic now makes sense

Reply:
related to this, we enforce 10 digits for all account numbers in the FiatDetailsScreen, but should it be country specific? Looks like per the spec, this is only for NG accounts https://github.com/fiatconnect/specification/blob/main/fiatconnect-api.md#8321-accountnumber

Reply:
good point.. should do this in a followup

Reply:
https://linear.app/valora/issue/ACT-357/account-number-length-should-be-country-specific